### PR TITLE
Added Enum Class for Flow Control

### PR DESCRIPTION
**Pull request by VFrancescon**
A sensor I am using requires flow control on, but the rest of the system does not.

Therefore I implemented an overloaded constructor and enum class to set it as required.

I tried my best to follow the format of the code as it already was, please forgive any lack of form.

Still new to pull requests and proper practices.

### CODE REVIEW

File: src/SerialPort.cpp [hunk 1]
```diff
@@ -209,7 +220,22 @@ namespace CppLinuxSerial {
 				THROW_EXCEPT("numStopBits_ value not supported!");
 		}
 
-		tty.c_cflag     &=  ~CRTSCTS;       // Disable hadrware flow control (RTS/CTS)
+		switch(flowControl_){
```
Comment (gbmhunter) [inline src/SerialPort.cpp:223]: What about FlowControl::SOFTWARE in this switch statement?

File: src/SerialPort.cpp [hunk 2]
```diff
@@ -414,7 +440,16 @@ namespace CppLinuxSerial {
 
 		//======================== (.c_iflag) ====================//
 
-		tty.c_iflag     &= ~(IXON | IXOFF | IXANY);			// Turn off s/w flow ctrl
+		switch(flowControl_){
```
Comment (gbmhunter) [inline src/SerialPort.cpp:443]: What about FlowControl::HARDWARE in this switch statement? Presumably you'd want to disable software flow control just like in the case of NONE.

**Review by gbmhunter — COMMENTED** [review]
Hi! Thanks for helping out this this open-source library! I have some comments for you on this PR.

File: src/SerialPort.cpp (hunk 2 above)
Comment (VFrancescon) [reply src/SerialPort.cpp:443]: Both of the two points here follow the same logic, so I'll reply here only.

Reading the codebase I found that the Hardware Flow Control (c_cflag) bit was toggled around line 200, with the Software Control (c_iflag) around line 400 (all in SerialPort.cpp).

While I did think to consolidate them all together, I kept them to two separate switch statements to maintain the structure already in place.

That being said, there would be no issue to group them up if we think that is more appropriate.

**Pull request merged**